buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java

```diff
@@ -323,6 +328,56 @@ public static String javaVersion(JavaVersion version) {
     return optionBuilder.build();
   }
 
+  private List<String> readKotlinCompilerArguments() {
+    try {
+      // Note: this is bad juju on Gradle 5.0, which would prefer you get the provider and lazily
+      // eval.
+      // We don't differentiate between test and non-test right now because Android projects don't
+      // support test-only configuration. Non-android projects theoretically can, but let's wait for
+      // someone to ask for that support first as it's not very common.
+      Optional<KotlinCompile> kotlinCompileTask = getProject()
+          .getTasks()
+          .withType(KotlinCompile.class)
+          .stream()
+          .findFirst();
```

Comment:
we might want to find the right task using variant info. let me see how we can do that. 
```
[task ':kotlin-app:compileDebugAndroidTestKotlin', task ':kotlin-app:compileDebugKotlin', task ':kotlin-app:compileDebugUnitTestKotlin', task ':kotlin-app:compileReleaseKotlin', task ':kotlin-app:compileReleaseUnitTestKotlin', task ':kotlin-app:kaptGenerateStubsDebugAndroidTestKotlin', task ':kotlin-app:kaptGenerateStubsDebugKotlin', task ':kotlin-app:kaptGenerateStubsDebugUnitTestKotlin', task ':kotlin-app:kaptGenerateStubsReleaseKotlin', task ':kotlin-app:kaptGenerateStubsReleaseUnitTestKotlin']

[task ':libraries:kotlinandroidlibrary:compileCustomKotlin', task ':libraries:kotlinandroidlibrary:compileCustomUnitTestKotlin', task ':libraries:kotlinandroidlibrary:compileDebugAndroidTestKotlin', task ':libraries:kotlinandroidlibrary:compileDebugKotlin', task ':libraries:kotlinandroidlibrary:compileDebugUnitTestKotlin', task ':libraries:kotlinandroidlibrary:compileReleaseKotlin', task ':libraries:kotlinandroidlibrary:compileReleaseUnitTestKotlin']

[task ':libraries:kotlinlibrary:compileKotlin', task ':libraries:kotlinlibrary:compileTestKotlin', task ':libraries:kotlinlibrary:kaptGenerateStubsKotlin', task ':libraries:kotlinlibrary:kaptGenerateStubsTestKotlin']
```
these are the different tasks I got